lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -548,6 +559,110 @@ func (c *ContainerProxy) createNewAttachClientWithTimeouts(connectTimeout, respo
 	return plClient, transport
 }
 
+func (c *ContainerProxy) Resize(vc *viccontainer.VicContainer, height, width int32) error {
+	defer trace.End(trace.Begin(vc.ContainerID))
+
+	if c.client == nil {
+		return derr.NewErrorWithStatusCode(fmt.Errorf("ContainerProxy failed to create a portlayer client"),
+			http.StatusInternalServerError)
+	}
+
+	plResizeParam := interaction.NewContainerResizeParamsWithContext(ctx).
+		WithID(vc.ContainerID).
+		WithHeight(height).
+		WithWidth(width)
+
+	_, err := c.client.Interaction.ContainerResize(plResizeParam)
+	if err != nil {
+		if _, isa := err.(*interaction.ContainerResizeNotFound); isa {
+			return ResourceNotFoundError(vc.ContainerID, "interaction connection")
+		}
+
+		// If we get here, most likely something went wrong with the port layer API server
+		return InternalServerError(err.Error())
+	}
+
+	return nil
+}
+
+// attacheStreams takes the the hijacked connections from the calling client and attaches
+// them to the 3 streams from the portlayer's rest server.
+// clStdin, clStdout, clStderr are the hijacked connection
+func (c *ContainerProxy) AttachStreams(ctx context.Context, vc *viccontainer.VicContainer, clStdin io.ReadCloser, clStdout, clStderr io.Writer, ca *backend.ContainerAttachConfig) error {
+	defer clStdin.Close()
+
+	// Cancel will close the child connections.
+	ctx, cancel := context.WithCancel(ctx)
+
+	var wg sync.WaitGroup
+	errors := make(chan error, 3)
+
+	// For stdin, we only have a timeout for connection.  There can be a long duration before
+	// the first entry so there is no timeout for response.
+	plClient, transport := createNewAttachClientWithTimeouts(attachConnectTimeout, 0, attachAttemptTimeout)
+	defer transport.Close()
+
+	if ca.UseStdin {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := copyStdIn(ctx, plClient, vc, clStdin, ca.DetachKeys)
+			if err != nil {
+				log.Errorf("container attach: stdin (%s): %s", vc.ContainerID, err.Error())
+			} else {
+				log.Infof("container attach: stdin (%s) done: %s", vc.ContainerID)
+			}
+
+			cancel()
+			errors <- err
+		}()
+	}
+
+	if ca.UseStdout {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := copyStdOut(ctx, plClient, attachAttemptTimeout, vc, clStdout)
+			if err != nil {
+				log.Errorf("container attach: stdout (%s): %s", vc.ContainerID, err.Error())
+			} else {
+				log.Infof("container attach: stdout (%s) done: %s", vc.ContainerID)
+			}
+
+			cancel()
+			errors <- err
+		}()
+	}
+
+	if ca.UseStderr {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := copyStdErr(ctx, plClient, vc, clStderr)
+			if err != nil {
+				log.Errorf("container attach: stderr (%s): %s", vc.ContainerID, err.Error())
+			} else {
+				log.Infof("container attach: stderr (%s) done: %s", vc.ContainerID)
+			}
+
+			cancel()
+			errors <- err
+		}()
+	}
+
+	// Wait for all stream copy to exit
+	wg.Wait()
+
```

Comment:
This is not the same function and not reflects the new code, please use the function below